crates/compiler/str/src/lib.rs

```diff
@@ -1,4 +1,8 @@
-//! provides `Roc` styled collection [reference counting](https://en.wikipedia.org/wiki/Reference_counting).
+//! Provides `Roc` styled collection [reference counting](https://en.wikipedia.org/wiki/Reference_counting). 
+//! This means the collection may or may not be safe to mutate in-place, may or 
+//! may not be reference counted, and may or may not need to be freed when no 
+//! longer in use. Whether each of these is true for a given collection can be 
+//! determined by inspecting that collection at runtime.
 #![crate_type = "lib"]
```

Comment:
Hmm this still seems weird because the pre-existing lib.rs comments talk about "collections" in general but only `Str` is defined.
But the code doesn't do anything and isn't imported to any other Rust package. It's not mentioned as a dependency in any Cargo.toml file. So I think we can probably delete this package anyway.
Maybe that's best for another PR though.